Added support for IW416 and IW612 socs for WI-Fi shell application

abhinavnxp opened the pull request:
Added support for NW61x and IW416 socs for EVKC1060 board for Wi-Fi shell application.

jukkar commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,127 @@
+CONFIG_NXP_WIFI_DRIVER_TASK_PRIO=3
+
+# log
+CONFIG_WIFI_LOG_LEVEL_DBG=y
```
I am not sure that the user of this board would want to automatically enable wifi debugging. This option is something that is set by the end user and not set by default.

abhinavnxp replied:
Made changes and split conf file into separate debug and module specific.

jukkar commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,127 @@
+
+# log
+CONFIG_WIFI_LOG_LEVEL_DBG=y
+CONFIG_THREAD_LOCAL_STORAGE=y
+CONFIG_ASSERT=y
+CONFIG_ASSERT_LEVEL=2
+CONFIG_LOG_MODE_IMMEDIATE=y
```
Similar comment to these, why do we need to set these.

abhinavnxp replied:
Made changes and split conf file into separate debug and module specific.

jukkar commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,127 @@
+CONFIG_LOG_MODE_IMMEDIATE=y
+
+# os
+CONFIG_THREAD_CUSTOM_DATA=y
+CONFIG_EVENTS=y
+CONFIG_SYS_HEAP_AUTO=y
+CONFIG_SCHED_MULTIQ=y
+CONFIG_ZVFS_OPEN_MAX=30
```
And here too. If some setting is needed that the sample will run, it should be set in the prj.conf. The board config file should only contains board specific changes.

abhinavnxp replied:
Made changes and split conf file into separate debug and module specific.

jukkar commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,127 @@
+CONFIG_WIFI_MGMT_AP_MAX_NUM_STA=8
+CONFIG_SHELL_CMD_BUFF_SIZE=512
+
+# net
```
Similar comment to the remaining options, why would we need to set them here as they seem to be very generic and not board specific options. It seems that you are trying to optimize this config file to your specific use case. That is a wrong approach, you can always specify all the application specific config options when compiling the sample.

abhinavnxp replied:
Made changes and split conf file into separate debug and module specific.

DerekSnell commented on boards/nxp/mimxrt1060_evk/mimxrt1060_evk.dtsi:
Need to update Copyright year:
```
Copyright 2018,2024-2025 NXP
```

abhinavnxp replied:
Done

DerekSnell commented on boards/nxp/mimxrt1060_evk/mimxrt1060_evk.dtsi:
```diff
@@ -232,6 +232,10 @@ zephyr_udc0: &usb1 {
 		disk-name = "SD";
 		status = "okay";
 	};
+	nxp_wifi {
+		compatible = "nxp,wifi";
+		status = "okay";
+	};
```
By default, this MIMXRT1060-EVK does not have Wi-Fi out-of-box, and I think we should not enable in this board file.  You could enable the Wi-Fi M.2 module as a shield, and then enable the `nxp_wifi` node in those shield files.

abhinavnxp replied:
Added shield

DerekSnell commented on boards/nxp/mimxrt1060_evk/mimxrt1060_evk_mimxrt1062_qspi_C.overlay:
Need to update the Copyright year

abhinavnxp replied:
Done

DerekSnell commented on drivers/wifi/nxp/Kconfig.nxp:
need to update copyright year

DerekSnell commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,127 @@
+CONFIG_WIFI_NXP=y
+CONFIG_NXP_IW416=y
+#CONFIG_NXP_IW61X=y
+CONFIG_NXP_WIFI_SHELL=y
+CONFIG_NXP_IW416_MURATA_1XK_M2=y
+#CONFIG_NXP_IW612_MURATA_2EL_M2=y
```
By default, this board MIMXRT1060-EVKC does not include the IW416 or IW61x SOCs.  Adding this board file `mimxrt1060_evk_mimxrt1062_qspi_C.conf` in this sample is misleading, as it indicates this board supports Wi-Fi.  I think it would be better to add a shield for the M.2 Wi-Fi module.  And then these Kconfig settings can be included in those shield files.  

This is probably true for most of the other Wi-Fi related Kconfigs in this file.

abhinavnxp replied:
Added shield

abhinavnxp commented on samples/net/wifi/shell/boards/mimxrt1060_evk_mimxrt1062_qspi_C.conf:
```diff
@@ -0,0 +1,126 @@
+CONFIG_NXP_WIFI_SHELL=y
+CONFIG_NXP_IW416_MURATA_1XK_M2=y
+#CONFIG_NXP_IW612_MURATA_2EL_M2=y
+
```
Hi @jukkar ,
I guess you are referring to old file which is no longer used. We have added module specific conf files and it contains only those configurations which are needed for NXP controllers to run smoothly. There is no specific application specific config added into those files. The macros which you are highlighting are the nxp controller macros and are needed by default for wifi shell app to run.
To simplify further I will remove old commit and try to accommodate everything into one single commit.

abhinavnxp replied:
Please check now.

abhinavnxp replied:
Hi @jukkar ,
As per discussion I have update the PR please review and confirm changes.

abhinavnxp committed:
boards: nxp: mimxrt1060_evk: Added support for IW612 and IW416 SoCs

- Added separate overlay conf files for wifi IW416 and IW612 SoCs.
- Added overlay debug conf file for nxp wifi needed to enable debug logs.
- Added additional configurations for module's region selection.

Signed-off-by: Abhinav Kulkarni <abhinav.kulkarni@nxp.com>

abhinavnxp commented on boards/nxp/mimxrt1060_evk/mimxrt1060_evk.dtsi:
```diff
@@ -1,5 +1,5 @@
 /*
```
Done please check

krish2718 commented:
> Hi @jukkar @krish2718 , Please help me with review and let me know why twister builds are failing. I will make changes if needed.

You can check twister failures here: https://github.com/zephyrproject-rtos/zephyr/actions/runs/14099115932/job/39495618118?pr=85749, looks like a wrong path:

```
CMake Error at /__w/zephyr/zephyr/modules/hal_nxp/mcux/mcux-sdk-ng/middleware/CMakeLists.txt:45 (add_subdirectory):
  add_subdirectory given source
  "/__w/zephyr/modules/hal/nxp/mcux/mcux-sdk-ng/middleware/usb" which is not
  an existing directory.
```

jukkar reviewed (dismissed):
LGTM

MaochenWang1 commented on samples/net/wifi/shell/nxp/overlay-nxp-wifi-IW416.conf:
```diff
@@ -0,0 +1,57 @@
+CONFIG_SCHED_MULTIQ=y
+
+# AP shell
+CONFIG_WIFI_SHELL_MAX_AP_STA=8
```
Please add `CONFIG_NXP_WIFI_SOFTAP_SUPPORT=y`, as `NXP_WIFI_SOFTAP_SUPPORT` will not enabled default

abhinavnxp replied:
Done

ehughes commented:
@abhinavnxp  

Hello!  The WiFi sample doesn't build as it can't find the firmware.   I think a node is required to enable WiFi as all of the kconfig settings depend on it.

![image](https://github.com/user-attachments/assets/882d9c22-b7a8-4909-a725-82f8b0fd4fbe)


 Is there a shield overlay that hasn't been added to the repo?


Could you provide some instructions in the readme.md

decsny commented:
@jukkar can you confirm if the configurations are still OK with you, the NXP HAL PR associated with this one finally merged so it should be ready for final review

decsny commented on samples/net/wifi/shell/nxp/overlay_hosted_mcu.conf:
```diff
@@ -0,0 +1,54 @@
+CONFIG_NET_DHCPV4_SERVER_ADDR_COUNT=32
+CONFIG_NET_IPV4=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_ZPERF=y
```
check line 24
